src/main/java/ToDoListFeature/ToDoList.java

```diff
@@ -33,83 +52,149 @@ private void unMarkTask(int index) {
-                int numberPosition = inputText.indexOf(" ");
-                markTask(Integer.parseInt(inputText.substring(numberPosition + 1)));
-            } else if(matchesPattern(inputText, UNMARK_FEATURE_PATTERN)) {
-                int numberPosition = inputText.indexOf(" ");
-                unMarkTask(Integer.parseInt(inputText.substring(numberPosition + 1)));
-            } else {
-                switch (inputText) {
-                case "list":
-                    printToDoList();
-                    break;
-
-                case "^mark \\d+$":
-                    System.out.println("nice");
-                    break;
-
-                case "help":
-                    featureIntroMessage();
-                    break;
-
-                case "quit":
-                    return;
-
-                default:
-                    addItem(new ToDoItem(inputText));
-                    break;
+            try {
+                if (!processKeywordInput(inputText)) {
+                    //add a specific todoTask based on the keyword input
+                    switch (inputText) {
+                    case "list":
+                        printToDoList();
+                        break;
+                
+                    case "help":
+                        featureIntroMessage();
+                        break;
+                
+                    case "quit":
+                        return;
+                
+                    default:
+                        System.out.println("invalid command, use [help] to show all the commands");
+                        break;
+                    }
                 }
+                messageDivider();
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("invalid command, use [help] to show all the commands");
+                messageDivider();
             }
         }
     }
 
+    /**
+     * Print the ToDoList
+     */
     public void printToDoList() {
         System.out.println("Here are the tasks in your list:");
         System.out.println(toString());
-        messageDivider();
     }
 
-    public void addItem(ToDoItem item) {
+    /**
+     * Add a ToDo item to the ToDoList
+     * @param item the ToDo item to be added
+     */
+    public void addItem(ToDo item) {
         this.toDoListArray.add(item);
         System.out.println("added: " + item.getName());
-        messageDivider();
+        System.out.println("Now you have " + this.toDoListArray.size() + " task(s) in the list");
     }
 
+    /**
+     * Get the string representation of the ToDoList
+     * @return the string representation of the ToDoList
+     */
     @Override
     public String toString() {
         if (this.toDoListArray.isEmpty()) {
             return "Empty ToDoList";
         }
         String output = "";
         int count = 1;
-        for (ToDoItem item: this.toDoListArray) {
+        for (ToDo item: this.toDoListArray) {
```

Comment:
> for (ToDo item: this.toDoListArray) {

Perhaps leave a space before the colon as it is a more conventional coding style, like this:
`for (ToDo item : this.toDoListArray)`

Reference:
[Google Java Style Guide - 4.6.2 Horizontal whitespace: Point number 4 ](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace)